api/gateway-operator/v2alpha1/controlplane_types.go

```diff
@@ -458,3 +463,82 @@ func (c *ControlPlane) SetConditions(conditions []metav1.Condition) {
 func (c *ControlPlane) GetExtensions() []commonv1alpha1.ExtensionRef {
 	return c.Spec.Extensions
 }
+
+// ControlPlaneKonnectOptions defines the Konnect-related configuration options for the ControlPlane.
+//
+// +apireference:kgo:include
+type ControlPlaneKonnectOptions struct {
+	// ConsumersSync defines the configuration for synchronizing consumers with Konnect.
+	//
+	// +optional
+	ConsumersSync *ControlPlaneKonnectConsumersSync `json:"consumersSync,omitempty"`
```

Comment:
Are we going to configure more options on syncing consumers with Konnect? If not, I think just `enabled/disabled` for `consumerSync` is enough.